### PR TITLE
Migrate Formstack Baton Lambda build process to Github Actions

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
This migrates the deploy process from TeamCity to Github actions.

## How to test
Performing a push to check if RiffRaff detects the update, followed by deploying to the CODE environment.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: .github/workflows/deploy-formstack.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+env:
+  DIRECTORY_NAME: formstack-baton-requests
+
+jobs:
+  CI:
+    name: Deploy formstack-baton-requests
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
```
Comment (kelvin-chappell) [inline .github/workflows/deploy-formstack.yml:16]: Perhaps add a note to say why these permissions are needed

File: .github/workflows/deploy-formstack.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'corretto'
+          cache: 'sbt'
+      - name: Build App
+        working-directory: ${{env.DIRECTORY_NAME}}
+        run: sbt clean compile riffRaffUpload
+      - name: Upload to Riff Raff
```
Comment (kelvin-chappell) [inline .github/workflows/deploy-formstack.yml:32]: Doesn't `riffRaffUpload` in the previous line upload to Riffraff?  I'm asking as I've never used `actions-riff-raff` so I'm a bit confused

File: .github/workflows/deploy-formstack.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+jobs:
+  CI:
+    name: Deploy formstack-baton-requests
+    runs-on: ubuntu-latest
+    # Allow GitHub to request an OIDC JWT ID token, to use with aws-actions/configure-aws-credentials
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
```
Comment (Mark-McCracken) [inline .github/workflows/deploy-formstack.yml]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by shtukas — DISMISSED** [review]

File: .github/workflows/deploy-formstack.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: aws-actions/configure-aws-credentials@v1
```
Comment (Mark-McCracken) [inline .github/workflows/deploy-formstack.yml]:
```suggestion
      - uses: aws-actions/configure-aws-credentials@v2
```

File: .github/workflows/deploy-formstack.yml (hunk 2 above)
Comment (Mark-McCracken) [reply .github/workflows/deploy-formstack.yml:32]: If it were me, I'd just leave off the `riffRaffArtifactResources`, and then get rid of the sbt plugin for riff raff.

**Review by Mark-McCracken — DISMISSED** [review]
Just a few tiny points to note, but these 2 version things don't actually have an impact if I remember correctly.
The riffRaffArtifactResources, I leave it up to you how you wanna deal with that.
But this would be fine to merge as it is, or if the only changes you make are around those 3 areas, feel free to merge with future commits without further approval.

File: .github/workflows/deploy-formstack.yml (hunk 2 above)
Comment (kelvin-chappell) [reply .github/workflows/deploy-formstack.yml:32]: Ah I see.  So they both do the same thing but one's through SBT and the other's through GHA.

**Review by Mark-McCracken — APPROVED** [review]

File: .github/workflows/deploy-formstack.yml (hunk 2 above)
Comment (Mark-McCracken) [reply .github/workflows/deploy-formstack.yml:32]: Yeah exactly. Little bit of a faff in github actions if you're not aware of whether it needs to produce a .jar or .deb file, and work out where it needs to land in the riffRaff folder structure, but hopefully once converted from TC to GHA, it'll be more clear to those who come to a project afresh

**Pull request merged**